src/Tasks/SystemState.cs

```diff
@@ -607,21 +615,19 @@ private string[] GetDirectories(string path, string pattern)
         /// <returns>True if the file exists.</returns>
         private bool FileExists(string path)
         {
-            /////////////////////////////////////////////////////////////////////////////////////////////
-            // FIRST -- Look in the primary cache for this path.
-            /////////////////////////////////////////////////////////////////////////////////////////////
-            object flag = instanceLocalFileExists[path];
-            if (flag != null)
+            if (instanceLocalLastModifiedCache.TryGetValue(path, out DateTime cachedLastModified))
             {
-                return (bool)flag;
+                return IsExistingFileTime(cachedLastModified);
             }
+            DateTime lastModified = getLastWriteTime(path);
+            instanceLocalLastModifiedCache[path] = lastModified;
+            return IsExistingFileTime(lastModified);
+        }
 
-            /////////////////////////////////////////////////////////////////////////////////////////////
-            // SECOND -- fall back to plain old File.Exists and cache the result.
-            /////////////////////////////////////////////////////////////////////////////////////////////
-            bool exists = fileExists(path);
-            instanceLocalFileExists[path] = exists;
-            return exists;
+        private bool IsExistingFileTime(DateTime lastModified)
```

Comment:
Don't really like the name of this method. It sounds like it's getting the existing file time, not does the time indicate it exists. But I don't have a great alternative :)

Reply:
@rainersigwald you're good at naming stuff, any thoughts?

Reply:
`bool FileTImestampIndicatesFileExists(string lastModified)`?

Reply:
I went with that. Only other one I can think of is flipping the condition and doing ```IsMinFileTime()```, clearer what it's measuring, but less clear that it's determining file existence.